app/controllers/coronavirus_form/check_answers_controller.rb

```diff
@@ -24,9 +26,114 @@ def submit
 
 private
 
+  helper_method :items
+
   def reference_number
     timestamp = Time.zone.now.strftime("%Y%m%d-%H%M%S")
     random_id = SecureRandom.hex(3).upcase
     "#{timestamp}-#{random_id}"
   end
+
+  def previous_path
+    contact_details_path
+  end
+
+  def items
+    @items ||= begin
+      questions.map { |question|
+        # We have answers as strings and hashes. The hashes need a little more
+        # work to make them readable.
+
+        if question.eql?("additional_product")
+          next add_extra_product(question)
+        end
+
+        if question.eql?("product_details")
+          next product_details(session[question])
+        end
+
+        value = case session[question]
+                when Hash
+                  concat_answer(session[question], question)
+                when Array
+                  session[question].flatten.to_sentence
+                else
+                  session[question]
+                end
+
+        [{
+          field: t("coronavirus_form.questions.#{question}.title"),
+          value: sanitize(value),
+          edit: {
+            href: question.dasherize,
```

Comment:
I suggest we add a param to this to know how many people will click through this:
```suggestion
            href: "#{question.dasherize}?change-answer",
```